code/VersionFeed.php

```diff
@@ -52,24 +53,28 @@ public function getDiffedChanges($highestVersion = null, $fullHistory = true) {
 		foreach ($versions as $version) {
 			$changed = false;
 
+			// Check if we have something to compare with.
 			if (isset($previous)) {
-				// We have something to compare with.
-				$diff = $this->owner->compareVersions($version->Version, $previous->Version);
 
 				// Produce the diff fields for use in the template.
 				if ($version->Title != $previous->Title) {
+					$diffTitle = Diff::compareHTML($version->Title, $previous->Title);
+
 					$version->DiffTitle = new HTMLText();
 					$version->DiffTitle->setValue(
 						sprintf(
-							'<div><em>%s</em>' . $diff->Title . '</div>',
+							'<div><em>%s</em> ' . $diffTitle . '</div>',
 							_t('RSSHistory.TITLECHANGED', 'Title has changed:')
 						)
 					);
 					$changed = true;
 				}
+
 				if ($version->Content != $previous->Content) {
+					$diffContent = Diff::compareHTML($version->Content, $previous->Content);
+
```

Comment:
Ah I know what's going on. `$diffContent` is set on the HTMLText, which is then rendered by the feed template, so shortcodes and all other transormations are interpolated further down the stream. The differencer will not split shortcodes - it only shows changes down to the html tag level.

Avoiding these two `forTemplate` calls gives us 25% performance gain (tested on 100 small changes).